src/System.Slices/src/System/Span.cs

```diff
@@ -299,27 +299,17 @@ public bool Equals(Span<T> other)
                 return false;
             }
 
-            Enumerator thisIt = this.GetEnumerator();
-            Enumerator otherIt = other.GetEnumerator();
-            while (true)
+            var nonBoxingComparer = EqualityComparer<T>.Default;
```

Comment:
@KrzysztofCwalina @krwq I think that it is unusual to have collection's Equal and GetHashCode method comparing and being dependent on the content. Maybe we should implement some SequenceEqual extension methods? One with the memory comparision for structs and other with IEqutable<T> constraint?